test/jdk/java/awt/Frame/FrameStateTest/FrameStateTest.java

```diff
@@ -41,419 +30,301 @@ called before setVisible(true) the Frame would be shown in NORMAL
+    CheckboxGroup cbgState = new CheckboxGroup();
+    CheckboxGroup cbgResize = new CheckboxGroup();
+    Checkbox cbIconState = new Checkbox("Frame state ICONIFIED", cbgState, true);
+    Checkbox cbNormState = new Checkbox("Frame state NORMAL", cbgState, false);
+    Checkbox cbNonResize = new Checkbox("Frame non-resizable", cbgResize, false);
+    Checkbox cbResize = new Checkbox("Frame Resizable", cbgResize, true);
+
+    CreateFrame icontst;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame
+                .builder()
+                .title("GetBoundsResizeTest Instructions")
+                .instructions(INSTRUCTIONS)
+                .testTimeOut(10)
+                .rows(25)
+                .columns(70)
+                .logArea(10)
+                .splitUIBottom(() -> new FrameStateTest().createPanel())
+                .build()
+                .awaitAndCheck();
+    }
+
+    public JPanel createPanel() {
+        JPanel panel = new JPanel();
+        panel.setLayout(new GridLayout(0, 3));
+        btnDispose.setEnabled(false);
+
+        panel.add(cbIconState);
+        panel.add(cbResize);
+        panel.add(btnCreate);
+        panel.add(cbNormState);
+        panel.add(cbNonResize);
+        panel.add(btnDispose);
+
+        btnDispose.addActionListener(this);
+        btnCreate.addActionListener(this);
 
+        return panel;
+    }
+
+    public void actionPerformed(ActionEvent evt) {
         if (evt.getSource() == btnCreate) {
             btnCreate.setEnabled(false);
             btnDispose.setEnabled(true);
-            icontst = new CreateFrame(iState, bResize);
-            icontst.show();
+            icontst =new CreateFrame(cbIconState.getState(), cbResize.getState());
+            icontst.setVisible(true);
         } else if (evt.getSource() == btnDispose) {
             btnCreate.setEnabled(true);
             btnDispose.setEnabled(false);
             icontst.dispose();
         }
     }
 
-    public void itemStateChanged(ItemEvent evt) {
-
-        if (cbNormState.getState()) iState = 0;
-        if (cbIconState.getState()) iState = 1;
-        if (cbResize.getState()) bResize = true;
-        if (cbNonResize.getState()) bResize = false;
+    static class CreateFrame extends Frame
+            implements ActionListener, WindowListener {
+
+        Button b1, b2, b3, b4, b5, b6, b7;
+        boolean isResizable;
+        String name = "Frame State Test";
+
+        CreateFrame(boolean iconified, boolean resizable) {
+            setTitle("Frame State Test (Window 2)");
+
+            isResizable = resizable;
+
+            PassFailJFrame.log("CREATING FRAME - Initially " +
+                    ((iconified) ? "ICONIFIED" : "NORMAL (NON-ICONIFIED)") + " and " +
+                    ((isResizable) ? "RESIZABLE" : "NON-RESIZABLE"));
+
+            setLayout(new FlowLayout());
+            add(b1 = new Button("resizable"));
+            add(b2 = new Button("resize"));
```

Comment:
@azvegint Minor: Should we consider renaming the "Resize" button to "Change Bounds"? Since when a non-resizable frame is created, the Resize option might seem confusing when we are changing the bounds of the frame.

Reply:
I am not sure about this, the button does exactly what its label says - resize the frame.

A non-resizable frame has always meant that it can't be resized by the user, but can be changed programmatically.

> java.awt.Frame#setResizable
Sets whether this frame is resizable by the user.

So I prefer to leave it as it is.